users/views.py

```diff
@@ -93,31 +97,17 @@ class UserSignUpSuccess(TemplateView):
     extra_context = {"MESSAGES": SuccessInfoAfterRegistrationText}
 
 
-class UserCreate(APIView):
-    renderer_classes = [renderers.TemplateHTMLRenderer]
+@method_decorator(login_required, name="dispatch")
+class UserCreate(CreateView):
     template_name = "user_create.html"
+    form_class = CustomUserCreationForm
 
-    @classmethod
-    def get(cls, request: HttpRequest) -> Response:
-        logger.info(f"User with id: {request.user.pk} entered UserCreate View")
-        serializer = UserCreateSerializer(context={"request": request})
-        return Response({"serializer": serializer})
+    def get_success_url(self) -> str:
+        return reverse("custom-users-list")
 
-    @classmethod
-    def post(cls, request: HttpRequest) -> Union[Response, HttpResponseRedirectBase]:
-        logger.debug(f"User with id: {request.user.pk} sent post on UserCreate View")
-        serializer = UserCreateSerializer(data=request.data)
-        if not serializer.is_valid():
-            logger.debug(f"Sent form is invalid due to those errors: {serializer.errors}")
-            return Response({"serializer": serializer, "errors": serializer.errors})
-        email = serializer.validated_data.get("email")
-        serializer.save()
-        user = CustomUser.objects.get(email=email)
-        user.set_password("passwduser")
-        user.full_clean()
-        user.save()
-        logger.info(f"New user with id {user.pk} has been created")
-        return redirect("custom-users-list")
+    def from_valid(self, form: CustomUserCreationForm) -> HttpResponseRedirect:
```

Comment:
I think this is not needed, as the super class is doing exactly that, but please check.

Reply:
It is - normally `form-valid` redirects to `DetailView` of newly created user. If we want to preserve the old behaviour (of redirecting to users list), this little method must stay as is ;)

Reply:
But the upper class is doing the same, please check:
https://github.com/django/django/blob/master/django/views/generic/edit.py#L55

And `save()` is done in different class which is also in the inheritance tree:
https://github.com/django/django/blob/master/django/views/generic/edit.py#L123